[markgcera] iP

markgcera opened the pull request:
### IRIS

> "Siri's _slightly_ less advanced little sister".

IRIS frees your mind of having to remember things you need to do. It's,

- CLI based
- Intuitive
- FAST ~~AF~~ to use

All you need to do is,

1. download it from [here](https://github.com/markgcera/ip).
2. run it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **NOT PAID**!

Features:

- [ ]  Managing tasks
- [ ]  Managing deadlines 
- [ ]  Reminders (maybe?)

Feel free to mess with IRIS. Here's the main method:

```
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input;
        ArrayList<Task> taskList = new ArrayList<>();

        Storage.loadTasksFromFile(taskList);

        Ui ui = new Ui();
        ui.printIntroMsg();

        input = scanner.nextLine();

        while (!input.equals("bye")) {
            ui.printSeparator();
            Parser.parse(input, taskList);
            Storage.updateTasksFile(taskList);
            ui.printSeparator();
            input = scanner.nextLine();
        }

        scanner.close();
        ui.printOutroMsg();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Complete Level 0

Commit:
Complete Level 1

Commit:
Complete Level 2

Commit:
Add Task class. Complete Level 3.

Commit:
Complete Level 4. Add Todo, Deadline, Event classes.

Commit:
Complete Level 5

Commit:
Complete Level 6

Commit:
Add try catch block

Commit:
Add test cases

Commit:
Remove end of file blank lines

Commit:
Use more OOP

Commit:
Complete Level 7

Commit:
Complete Level 8

Commit:
Complete A-MoreOOP

Commit:
Complete A-Packages.

Commit:
Complete A-Packages.

Commit:
Merge commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552'

* commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552':
  Bump gradle and lib version
  Add Gradle support

Commit:
Complete A-Gradle.

Commit:
Make basic JUnit tests

Commit:
Add JavaDoc comments

Commit:
Tweak code to conform to coding standard.

Commit:
Complete Level 9 Find.

Commit:
Merge branch 'branch-A-JavaDoc' into branch-A-CodingStandard

* branch-A-JavaDoc:
  Add JavaDoc comments

# Conflicts:
#	src/main/java/duke/Event.java

Commit:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

* branch-A-CodingStandard:
  Tweak code to conform to coding standard.
  Add JavaDoc comments

Commit:
Add JavaDoc to find method.

Commit:
Complete Level 10.

MagnificentCreature commented on src/main/java/GUI/MainWindow.java:
```diff
@@ -0,0 +1,52 @@
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+    }
+
+    public void setIris(Iris d) {
```
Does "d" mean anything?

MagnificentCreature commented on src/main/java/Tasks/Event.java:
```diff
@@ -0,0 +1,37 @@
+     */
+    public String getFromTo() {
+        return from.format(DateTimeFormatter.ofPattern("MMM d yyyy"))
+                + " to "
```
I like the spacing of " to " to a new line here, makes it easier to understand which parts are hard code printed and which ones are generated/varied

MagnificentCreature commented on src/main/java/Tasks/Event.java:
```diff
@@ -0,0 +1,37 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + "(from:"
```
Is there a reason why perhaps the "to:" here is not following the pattern above?

Commit:
More OOP.

Commit:
no message

Commit:
no message

MagnificentCreature commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,35 @@
+        case "mark":
+            return executor.markTask(inputParts, taskList, true);
+        case "unmark":
+            return executor.markTask(inputParts, taskList, false);
```
Nice switch case indentation

MagnificentCreature reviewed (commented):
Overall, I found your code easy to read for the most part. 

I couldn't find any significant flaws.

Commit:
Add checkstyle

cbj252 commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,62 @@
+     */
+    public void markUndone() {
+
+        done = false;
```
Inconsistent line breaks. markDone() has no line breaks in the function while markUndone() does. Not a code violation, but a bit strange.

cbj252 commented on src/main/java/workers/MarkWorker.java:
```diff
@@ -0,0 +1,29 @@
+ */
+public class MarkWorker extends TaskWorker {
+    /**
+     * This method marks the Task at the specified index as done.
```
Javadoc comments should have an empty line between description and parameter section.

cbj252 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,86 @@
+     * This method updates the file by converting the Tasks into text.
+     * @param taskList
+     */
+    public void updateTasksFile(ArrayList<Task> taskList) {
```
Could this be a clearer verb like updateTasksToFile?

cbj252 commented on src/main/java/workers/AddWorker.java:
```diff
@@ -0,0 +1,50 @@
+                newTask = new Deadline(commandParts[0], commandParts[1]);
+            } else if (inputParts[0].equals("event")) {
+                String[] commandParts = description.split("/from ", 2);
+                String[] eventParts = commandParts[1].split("/to ");
```
Like the choice of "inputParts", "commandParts" and "eventParts" as names - plural shows they are arrays, the common naming scheme reflects that they are all parts of the input and they are all representative of what they contain.

cbj252 reviewed (commented):
Well written overall.

Commit:
Add assert statements.

No assert statements as of previous commits. There is a need to validate that there are no programmer errors.

Add assert statements to multiple classes to check if function arguments are correct, and if the outputs are correct.

Commit:
Storage class' updateTasksToFile method is unnecessarily changing the dates to another format. This causes clutter.

Bug is found regarding the Event tasks in that the from and to dates are being updated into the file as one string without a divider.
This causes a problem when trying to load the tasks from the file.

Let's change the storage of dates into the file without changing its DateTime format.

We can also change the from and to dates for Event tasks to be separated by a divider to allow the loadTasksFromFile to read the dates properly.

Commit:
AddWorker class is functional, but a little unreadable when it comes to reading them arguments that get passed through to the other methods.

Let's make the code a little bit more readable by adding intermediary variables, albeit they make the code slightly longer.

markgcera committed:
Merge pull request #1 from markgcera/branch-A-Assertions

Add assert statements.

markgcera committed:
Merge branch 'master' into branch-A-CodeQuality

markgcera committed:
Merge pull request #2 from markgcera/branch-A-CodeQuality

Branch a code quality

markgcera committed:
Revert "Branch a code quality"

markgcera committed:
Merge pull request #3 from markgcera/revert-2-branch-A-CodeQuality

Revert "Branch a code quality"

markgcera committed:
Revert "Revert "Branch a code quality""

markgcera committed:
Merge pull request #4 from markgcera/revert-3-revert-2-branch-A-CodeQuality

Revert "Revert "Branch a code quality""

Commit:
Refactor the methods to check if the output string is empty to make them more readable.
Refactor the if else statements in the AddWorker class to switch statements to make them more readable.

markgcera committed:
Merge pull request #5 from markgcera/branch-A-CodeQuality

Refactor the methods to check if the output string is empty to make t…

Commit:
Add CI to master branch.

Commit:
Test to see if the OAuth error is caused by content of the yml file or the yml file itself.

Commit:
no message

markgcera committed:
Create gradle.yml

Commit:
Add DuplicateDetector classes to detect if the Task to be added already exists.

Commit:
Add an opening message.

Commit:
Refactor the AddWorker class to add more abstraction to make the code cleaner.

Commit:
Extract out the duplicate tester and add error handling for it.

Commit:
Add more tests

Commit:
Add User Guide

Commit:
Add more test cases

Commit:
Add documentation

Commit:
Add IRIS name to User Guide

Commit:
Change title of User Guide

Commit:
Bug fix and checkstyle fix
